syntax = "proto3";

package proto;

service FlightsInfo {
    rpc AddFlight(NewFlight) returns (AddResponse);
    rpc ModifyFlight(UpdatedFlight) returns (ModFlightResponse);
    rpc ModifySeats(UpdatedSeats) returns (ModSeatsResponse);
    rpc ModifyServices(UpdatedServices) returns (ModServicesResponse);
    rpc GetPriceFlight(PriceRequest) returns (PriceReply);
    rpc GetAllSeatsFlight(SeatCostRequest) returns (stream SeatCostReply);
    rpc GetAlladditionalServicesFlight(AdditionalServiceCostRequest) returns (stream AdditionalServiceCostReply);
    //Logging
    rpc getLogFileMan(GetLogFileRequestMan) returns (stream GetLogFileReplyMan);
}

message SeatCostRequest {
    string compagnia = 1;
}

message SeatCostReply {
    string prezzo = 1;
}

message AdditionalServiceCostRequest {
    string compagnia = 1;
}

message AdditionalServiceCostReply {
    string prezzo = 1;
}

message PriceRequest {
    string idVolo = 1;
}

message PriceReply {
    string price = 1;
}

message NewFlight {
    string id = 1;
    string date = 2;
    string departureAirport = 3;
    string arrivalAirport = 4;
    string departureTime = 5;
    string arrivalTime = 6;
    string airline = 7;
    string price = 8;
    int32 seats = 9;
}

message AddResponse {
    bool isOk = 1;
    string error = 2;
}

message UpdatedFlight {
    string flightId = 1;
    string newPrice = 2;
    string airline = 3;
}

message ModFlightResponse {
    bool isOk = 1;
    string error = 2;
}

message UpdatedSeats {
    string airline = 1;
    string price1 = 2;
    string price2 = 3;
    string price6 = 4;
    string price16 = 5;
    string price18 = 6;
}

message ModSeatsResponse {
    bool isOk = 1;
    string error = 2;
}

message UpdatedServices {
    string airline = 1;
    string priceBM = 2;
    string priceBG = 3;
    string priceBS = 4;
    string priceAD = 5;
    string priceAB = 6;
    string priceTN = 7;
}

message ModServicesResponse {
    bool isOk = 1;
    string error = 2;
}

message GetLogFileRequestMan {
    int32 numRichiesta = 1;
}

message GetLogFileReplyMan {
    bytes chunk_file  = 1;
    int32 num_chunk = 2;
}
